src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -0,0 +1,454 @@
+
+			if (!RChannel || !GChannel || !BChannel)
+				assert(0);
+
+			if (AChannel)
+				doesItSupportAlphaChannel = true;
+			else
+				doesItSupportAlphaChannel = false;
+
+			auto doesRGBFormatHaveTheSameFormatLikePassedToIt = [&](const PixelType ImfTypeToCompare)
+			{
+				return (RChannel->type == ImfTypeToCompare && GChannel->type == ImfTypeToCompare && BChannel->type == ImfTypeToCompare);
+			};
+
+			auto doesAChannelExistAndIsTheSameFormatLikePassedToIt = [&](const PixelType ImfTypeToCompare)
+			{
+				return (AChannel && AChannel->type == ImfTypeToCompare);
+			};
+
+			if (doesRGBFormatHaveTheSameFormatLikePassedToIt(PixelType::HALF))
+			{
+				if (doesAChannelExistAndIsTheSameFormatLikePassedToIt(PixelType::HALF))
+					format = EF_R16G16B16A16_SFLOAT;
+				else
+					format = EF_R16G16B16_SFLOAT;
+			}
+			else if (doesRGBFormatHaveTheSameFormatLikePassedToIt(PixelType::FLOAT))
+			{
+				if (doesAChannelExistAndIsTheSameFormatLikePassedToIt(PixelType::FLOAT))
+					format = EF_R32G32B32A32_SFLOAT;
+				else
+					format = EF_R32G32B32_SFLOAT;
+			}
+			else if (doesRGBFormatHaveTheSameFormatLikePassedToIt(PixelType::UINT))
+			{
+				if (doesAChannelExistAndIsTheSameFormatLikePassedToIt(PixelType::UINT))
+					format = EF_R32G32B32A32_UINT;
+				else
+					format = EF_R32G32B32_UINT;
+			}
+			else
+				assert(0);
+		}
+
+		bool readVersionField(io::IReadFile* _file, SContext& ctx)
+		{
+			RgbaInputFile file(_file->getFileName().c_str());
+			auto& versionField = ctx.versionField;
+			
+			versionField.mainDataRegisterField = file.version();
+
+			auto isTheBitActive = [&](uint16_t bitToCheck)
+			{
+				return (versionField.mainDataRegisterField & (1 << bitToCheck - 1));
+			};
+
+			versionField.fileFormatVersionNumber |= isTheBitActive(1) | isTheBitActive(2) | isTheBitActive(3) | isTheBitActive(4) | isTheBitActive(5) | isTheBitActive(6) | isTheBitActive(7) | isTheBitActive(8);
+
+			if (!isTheBitActive(11) && !isTheBitActive(12))
+			{
+				versionField.Compoment.type = SContext::VersionField::Compoment::SINGLE_PART_FILE;
+
+				if (isTheBitActive(9))
+					versionField.Compoment.singlePartFileCompomentSubTypes = SContext::VersionField::Compoment::TILES;
+				else
+					versionField.Compoment.singlePartFileCompomentSubTypes = SContext::VersionField::Compoment::SCAN_LINES;
+			}
+			else if (!isTheBitActive(9) && !isTheBitActive(11) && isTheBitActive(12))
+			{
+				versionField.Compoment.type = SContext::VersionField::Compoment::MULTI_PART_FILE;
+				versionField.Compoment.singlePartFileCompomentSubTypes = SContext::VersionField::Compoment::SCAN_LINES_OR_TILES;
+			}
+
+			if (!isTheBitActive(9) && isTheBitActive(11) && isTheBitActive(12))
+				versionField.doesItSupportDeepData = true;
+			else
+				versionField.doesItSupportDeepData = false;
+
+			if (isTheBitActive(10))
+				versionField.doesFileContainLongNames = true;
+			else
+				versionField.doesFileContainLongNames = false;
```

Comment:
if you dont support something here that would prevent loading a file, feel free to return false (like multi-part file)

Reply:
sure, to fix!